src/components/Chat/ChatDetails.tsx

```diff
@@ -35,174 +33,147 @@ interface ChatDetailsProperties {
     subscribable?: boolean;
 }
 
-interface ChatDetailsState {
-    channelId: string;
-    subscribable: boolean;
-    notify_unread: boolean;
-    notify_mentioned: boolean;
-}
+export function ChatDetails(props: ChatDetailsProperties): JSX.Element {
+    const channel = props.chatChannelId;
 
-export class ChatDetails extends React.PureComponent<ChatDetailsProperties, ChatDetailsState> {
-    constructor(props) {
-        super(props);
-        const channel = this.props.chatChannelId;
-        if (channel) {
-            this.state = {
-                channelId: channel,
-                subscribable: props.subscribable,
-                notify_unread: false,
-                notify_mentioned: false,
-            };
-        }
-    }
+    const [channelId, setChannelId] = React.useState(channel);
+    const [subscribable, setSubscribable] = React.useState(props.subscribable);
```

Comment:
Perhaps not an issue for this PR, but it's odd to me that subscribable is both a prop and a state variable...

It looks like there's a `useEffect` below that mirrors `props.subscribable` into `subscribable`, but it would be simpler and safer to just use `props.subscribable` directly.

Reply:
Yeah that's totally true.. it was a very mechanical port, would be good to further clean that up